src/test/java/org/apache/sling/resourceresolver/impl/mapping/ResourceMapperImplTest.java

```diff
@@ -355,6 +358,31 @@ public void mapNestedAliasTarget() {
                 .allMappingsWithRequest("/app/alias-parent/alias-child", "/app/alias-parent/child", "/app/parent/alias-child")
                 .verify(resolver, req);
     }
+    
+    /**
+     * Validates that mappings for an empty path
+     *
+     * @throws LoginException
+     */
+    @Test
+    public void mapEmptyPathWithNonExistingResource() {
+        MapEntriesHandler mapEntriesHandler = mock(MapEntriesHandler.class);
+        when(mapEntriesHandler.getVanityPathMappings()).thenReturn(Collections.emptyMap());
+
+        ResourceResolverImpl resolver = mock(ResourceResolverImpl.class);
+        when(resolver.resolveInternal(any(), any())).thenReturn(null);
```

Comment:
Can we make this something return `null` only for the root path and delegate to the real resolver? I think that's a more likely scenario.

Reply:
Apparently root path is not empty this was done to replicate the scenario which was observed in bug where path is empty and then resolveInternal ought to return null.